File: code_examples/3-Structured_Problem_Solving/11-Compoud_Boolean/Java/Main.java
/*
 * This program checks for a runtime error
 *
 * @author  Mr Coxall
 * @version 1.0
 * @since   2020-09-01
 */

import java.util.Scanner;

public class Main {
  public static void main(String[] args) {
    // this function checks for a runtime error

    // create Scanner object for user input
    Scanner scanner = new Scanner(System.in);

    // input
    System.out.print("Enter an integer: ");
    String integerAsString = scanner.nextLine();

    // process & output
    try {
      int integerAsNumber = Integer.parseInt(integerAsString);
      System.out.printf("You entered: %d.%n", integerAsNumber);
    } catch (NumberFormatException e) {
      System.out.printf("The error was: %s.%n", e.getMessage());
    } finally {
      System.out.println("Thanks for playing.");
    }

    // close the Scanner object
    scanner.close();
    System.out.println("\nDone.");
  }
}

File: code_examples/3-Structured_Problem_Solving/11-Compoud_Boolean/JavaScript/main.js
/* Created by: Mr. Coxall
 * Created on: Sep 2020
 * This program checks for a runtime error
 */

const prompt = require("prompt-sync")()

// input
const integerAsString = prompt("Enter an integer: ")

// process & output
try{
  const integerAsNumber = parseInt(integerAsString)
  console.log(`You entered: ${integerAsNumber}.`)
} catch (err) {
  console.log(`The error was: {err}.`)
} finally {
  console.log("Thanks for playing.")
}

console.log("\nDone.")
